node/libs/protobuf_build/src/lib.rs

```diff
@@ -52,70 +56,68 @@ fn traverse_files(
     Ok(())
 }
 
-/// Protobuf descriptor + info about the mapping to rust code.
+/// Manifest of a Protobuf compilation target containing information about the compilation process.
 #[derive(Debug)]
-pub struct Descriptor {
+struct Manifest {
     /// Root proto package that all proto files in this descriptor belong to.
-    /// Rust types have been generated relative to this root.
-    proto_root: ProtoName,
-    /// Raw descriptor proto.
-    descriptor_proto: prost_types::FileDescriptorSet,
-    /// Direct dependencies of this descriptor.
-    dependencies: Vec<&'static Descriptor>,
+    proto_root: ProtoPath,
+    /// Absolute path to the descriptor.
+    descriptor_path: String,
+    /// Tuples of `proto_root` and absolute paths to the corresponding descriptor for all dependencies
+    /// including transitive ones.
```

Comment:
including transitive deps is a tradeoff, it should be possible to avoid that.

Reply:
ok, given that these are just paths to the descriptors and the descriptors themselves are not copied this should be fine

Reply:
I've thought about it as well and considered against it 🙃 It would be possible by writing a manifest into a file somewhere in `$OUT_DIR` and passing around paths to these files. IMO, the downsides of this approach (an additional level of indirection) outweigh the benefits.